### DISCUSSION TITLE
Clipping a mesh to an isosurface

### DISCUSSION THREAD

**Discussion by stla** [CGAL/cgal#7546] [Ideas]
Hello,

In the R package **rgl**, there is a function to clip a 3d mesh to a surface defined by `f(x,y,z) > 0` where `f` is a given function. For example I've just done this tennis ball with this feature (I clipped the sphere two times, with `f` and with `-f`): 

![TennisBall](https://github.com/CGAL/cgal/assets/4466543/c361264e-b10f-4a25-a58d-3ced6044c056)

So we can achieve the same result with CGAL by using `PMP::clip` if we have a mesh of `f(x,y,z)>0`. It seems to me that the way used by **rgl** is more direct and more efficient. Maybe you would like to add this feature to CGAL? This is not complicated.

**Comment by MaelRL** [comment]
Do you mean something like https://github.com/CGAL/cgal/pull/5895?

**Reply by stla** [reply]
Hmm that's possible, I didn't understand everything yet. Is the bunny splitted in several pieces here?:

![bunny](https://github.com/CGAL/cgal/assets/4466543/d34aaacd-8fdc-4ec0-947e-ce5b759cd04e)

**Reply by stla** [reply]
What I mean is really equivalent to `PMP::clip(sphereMesh, isoMesh)`.
